classes/file/PKPLibraryFileManager.php

```diff
@@ -112,16 +112,49 @@ public function &copyFromTemporaryFile(&$temporaryFile, $libraryFileType)
         $libraryFileDao = DAORegistry::getDAO('LibraryFileDAO'); /** @var LibraryFileDAO $libraryFileDao */
         $libraryFile = $libraryFileDao->newDataObject();
 
+        $libraryFile = $this->assignFromTemporaryFile($temporaryFile, $libraryFileType, $libraryFile);
+        if (!$this->copyFile($temporaryFile->getFilePath(), $this->getBasePath() . $libraryFile->getServerFileName())) {
+            return false;
+        }
+
+        return $libraryFile;
+    }
+
+    /**
+     * Routine to replace a library file from a temporary file.
+     * @param $temporaryFile object
+     * @param $libraryFileType int LIBRARY_FILE_TYPE_...
+     * @param $libraryFile LibraryFile 
+     * @return LibraryFile|false the updated LibraryFile, or false on error
+     */
+    function &replaceFromTemporaryFile(&$temporaryFile, $libraryFileType, $libraryFile) {
```

Comment:
You're probably just following existing (very old) patterns, but the return reference here isn't necessary and should be removed.

Reply:
Also, not crucial, but for `stable-3_4_0` and newer, our preference is to use PHP8.0-compatible type hints for new code. For `main` it'll be PHP8.1-compatible. With that in place, `@param` tags are encouraged only when they add something more than the type hints do.